src/OmniSharp.Roslyn.CSharp/Services/Intellisense/IntellisenseService.cs

```diff
@@ -37,54 +37,60 @@ public IntellisenseService(OmniSharpWorkspace workspace, FormattingOptions forma
                         {
-                            // Note: For keywords, we'll just assume that the completion text is the same
-                            // as the display text.
-                            var keyword = item.DisplayText;
-                            if (keyword.IsValidCompletionFor(wordToComplete))
+                            var symbols = await item.GetCompletionSymbols(recommendedSymbols, document);
+                            if (symbols.Any())
                             {
-                                var response = new AutoCompleteResponse()
+                                foreach (var symbol in symbols)
                                 {
-                                    CompletionText = item.DisplayText,
-                                    DisplayText = item.DisplayText,
-                                    Snippet = item.DisplayText,
-                                    Kind = request.WantKind ? "Keyword" : null
-                                };
-
-                                completions.Add(response);
+                                    if (symbol != null)
+                                    {
+                                        if (request.WantSnippet)
+                                        {
+                                            foreach (var completion in MakeSnippetedResponses(request, symbol, item.DisplayText))
+                                            {
+                                                completions.Add(completion);
+                                            }
+                                        }
+                                        else
+                                        {
+                                            completions.Add(MakeAutoCompleteResponse(request, symbol, item.DisplayText));
+                                        }
+                                    }
+                                }
+
+                                // if we had any symbols from the completion, we can continue, otherwise it means
+                                // the completion didn't have an associated symbol so we'll add it manually
+                                continue;
                             }
-                        }
-                    }
-                }
 
-                var model = await document.GetSemanticModelAsync();
-                var symbols = await Recommender.GetRecommendedSymbolsAtPositionAsync(model, position, _workspace);
+                            // for other completions, i.e. keywords, create a simple AutoCompleteResponse
+                            // we'll just assume that the completion text is the same
+                            // as the display text.
+                            var response = new AutoCompleteResponse()
+                            {
+                                CompletionText = item.DisplayText,
```

Comment:
To support completion properly we need a new api similar to what LSP provides with `additionalTextEdits`?